### DISCUSSION TITLE
what tools do you use to create new app manifests for scoop?

### DISCUSSION THREAD

**Discussion by jmiguelhdez** [ScoopInstaller/Scoop#5829] [Help and Questions] [answered]
so i understand that scoop can extract msi and exe files, but some of the manifest require post installation scripts(powershell), register files(for file association), etc.

how do you know what the original install does to duplicate the functionality? is there a sandboxing tool or msi tool that tells you that?

i was thinking maybe process monitor might help but i wanted to get some feedback from people that have created manifests before. is there a known BKM/tutorial/documentation on how to analyze the app installer and create the scoop manifest?

**Comment by HUMORCE** [comment, marked as answer]
msi installer:
- View with `main/lessmsi`.
- Write manfiest directly then check the installation dir.

exe:
- Check with `main/7zip` to confirm if it can be decompressed directly.
- If NOT, use `extras/uniextract2` for further detection, they are usually created by Inno Setup.
- if NOT, analyze in VMs/Windows Sandbox using [TOTAL UNINSTALL](https://www.martau.com/) or ProcessMonitor(`extras/sysinternals`).

Normally, I do not write scripts for context menus, file associations, protocol associations, service creatation, etc. for apps. e.g. `main/7zip`: Even though its GUI settings already include a context menu creation option, someone still made a script for it. I can't understand that.

Some special manifests:
- `extras/autohotkey`: Mock its original installer to kept user's data(sub-versions of ahk), file associations etc.
- `extras/fastcopy`: Use its oirginal (un)installer to avoid update/uninstall error due to file lock(it's dll for context menu)
- `nuke/multipass`([link](https://github.com/HUMORCE/nuke/blob/master/bucket/multipass.json)): With self-use scripts for daemon management.

For apps that can not **Scooped**: Just give up and use its original insatller, e.g. Web browsers, Steam, vcredist, whatsapp, logitech g hub, photoshop, etc.

**Scooped**: Make the app as portable as possible. Of course, many GUI apps installed with Scoop are not fully portable, but they can be extracted and works just fine with Scoop environment or directory structure.

**Reply by jmiguelhdez** [reply]
thanks, this helps a lot
